functions/index.ts

```diff
@@ -245,3 +246,26 @@ functions.database.ref('gamePortal/gameSpec/reviews/{reviewedGameSpecId}/{review
       return starsSummary;
     });
 });
+
+exports.resizeImage =
+functions.storage.object().onChange.onChange(async (event:any) => {
+  console.log("Resizing the image");
+  // Extract file data
+  const path = event.data.name;
+  const bucket = event.data.bucket;
+  const [imgType, uid, id] = path.split("/");
+
+  if(imgType !== "upload") return;
+
+  // Download file into tmp directory
+  const tmpFilePath = `/tmp/${id}.jpg`;
+  const file = bucket.file(path);
+  console.log("Downloading image to: ", tmpFilePath);
+  await file.download({ destination : tmpFilePath});
+
+  // Resize it and upload it to cloud storage
+  console.log("Resizing image and uploading to cloud storage...");
+  await spawn('convert', [tmpFilePath, '-thumbnail', '900x900>', tmpFilePath]);
```

Comment:
Add a comment about what this convert does:
it calls the function convert from imagemagick and pass it some parameters, like -thumbnail , see doc here:
http://www.imagemagick.org/Usage/thumbnails/